### ISSUE TITLE
Suggestion for classification of the compositional constituants

### ISSUE THREAD

**Issue by maweber-bia** [Southern-Cross-Plant-Science/cdno#61]
As the coverage of CDNO is now "Compositional Dietary Nutrition Ontology" I would suggest to clarify the current hierarchy which is "dietary nutritional component"

Actually, they are a mismatch between compositional constituents and "real" nutritional components.

I mean we can enlarge the coverage of this branch by renaming it "compositional constituent" and then distinguish between nutrient subclasses and other  subclasses (see my suggestion below)

This is the direction I am taking in my own work. I would be happy if we could agree on this.

<img width="203" alt="composition-branch" src="https://user-images.githubusercontent.com/64576528/119254378-36a29a00-bbb6-11eb-8fe1-55d94662e33e.PNG">

**Comment by LilyAndres** [comment]
Thanks @maweber-bia, I will discuss this with the team and get back to you during the week. I can see some points that we could agree on and incorporate in CDNO.

**Comment by kaiiam** [comment]
> mismatch between compositional constituents and "real" nutritional components.

Thanks @maweber-bia for posting this. Would you mind elaborating what is different between "real" and compositional? That might help us better model it. Ontologies allow for multi-inheritance, aka terms having multiple parents, so perhaps it is possible to represent thing from multiple perspectives.

**Comment by kaiiam** [comment]
Keep in mind that we'll need to refactor the current dietary nutritional component hierarchy to remove any BFO role terms including vitamins and also plant secondary metabolite (which is also a role). It'd be best for this hierarchy to follow BFO and only contain independent continuants (entities).

**Comment by maweber-bia** [comment]
@kaiiam Sorry for misusing "real": actually, I mean something which is considered to be nutrients = nutritive components

I think that secondary plant metabolites should be considered as entities too (but perhaps there is the need to renamed the class as phytochemical to be more generic ?)

**Comment by kaiiam** [comment]
> Sorry for misusing "real": actually, I mean something which is considered to be a nutrient= nutritive component

I'm just trying to understand here are you suggesting something like differentiating chemicals which are nutrients (has some sort of role in nutrition) from those compositional which are eaten but don't provide some sort of nutrition? Is that what you mean if not could you elaborate more so we can try to better model it. 

> I think that secondary plant metabolites should be considered as entities too (but perhaps there is the need to have the class renamed as phytochemical to be more generic ?)

simply because `plant secondary metabolite` is a role we should be using that particular term. If something like `phytochemical` is a more generic and domain accepted term then I'd say we should use something like that instead. Either request it in CHEBI or create it in CDNO.

**Comment by kaiiam** [comment]
Edit in previous comment view it on github if using email.

**Comment by maweber-bia** [comment]
@kaiiam,  

my response to your comments

> I'm just trying to understand here are you suggesting something like differentiating chemicals which are nutrients (has some sort of role in nutrition) from those compositional which are eaten but don't provide some sort of nutrition? Is that what you mean if not could you elaborate more so we can try to better model it.

--> yes, I was trying to suggest "differentiating chemicals which are nutrients (has some sort of role in nutrition) from those compositional which are eaten but don't provide some sort of nutrition"



> If something like `phytochemical` is a more generic and domain accepted term then I'd say we should use something like that instead. 

--> please see the reference <https://pubmed.ncbi.nlm.nih.gov/31769838/>

**Comment by kaiiam** [comment]
Great thanks @maweber-bia! @CropStoreDb and @LilyAndres do you agree with the proposed change from `plant secondary metabolite` to `phytochemical` or similar? Does it align with the original nutrition framework CDNO was created from? 

I'm just the ontology engineer for the project, not a nutrition scientist, so I won't weigh in except to say that it'd good be too to have a generic high level representation of the intended concept preferably using terminology already accepted by the community. Also we **need to not** be using a BFO role descendant like `plant secondary metabolite` in the hierarchy and instead use or create a material entity descendant term.

**Comment by kaiiam** [comment]
> What we mentioned is that we were going to have plant secondary metabolite as a material entity and not as a role for now

Yes, your right I tough it was a role class but it's our own CDNO class [plant secondary metabolite](http://purl.obolibrary.org/obo/CDNO_0000013) so were good to keep it in the DNC hierarchy as it's a material entity. Sorry I missed that earlier disregard previous comments about needing to change it for the sake of making BFO correct.
> See my quote from #29

If we have a new term that's more general than just plants to replace or add to the DNC we should do that, (and maybe depreciate this if needed).

**Comment by LilyAndres** [comment]
@maweber-bia thanks a lot for all your comments. We are trying to check if the hierarchies that you mentioned work for us, is good to mention that we created the [CDNO framework](https://github.com/Southern-Cross-Plant-Science/cdno/blob/master/src/ontology/util/CDN-DF_v.2.0.xlsx), based on some information from the Food Composition Tables such as USDA, INFOODs and EuroFIR. For this reason, we might not be able to change the label from `'dietary nutritional component'` to `'component'`, as for us `'dietary nutritional component'` works. If we change it, we might need to contact Uberon as they already have [this term](https://www.ebi.ac.uk/ols/ontologies/uberon/properties?iri=http%3A%2F%2Fwww.geneontology.org%2Fformats%2FoboInOwl%23component) (not sure if this really exists as a term though) this is why we wanted to provide specific terms instead of general terms that have been defined in other ontologies unless is within the CDNO scope, then we can reuse them. 

Here is the definition of `'dietary nutritional component'` in CDNO:

"A **material entity** that is ingested and contributes to survival, growth, and development."

_____________________

For the other subclasses you mentioned. Are "nutrients", "micronutrients" and "macronutrients" considered as roles? If so, we probably want to describe them in the  `'dietary role/function'` class. Happy to discuss.

_____________________

I think with the change of Crop DNO to Compositional DNO, we might need to look at special terms we had for plants eg: 

1. deprecate `'dry plant matter'` [CDNO:0000002] and create a new term `'dry matter'`?
But checking an old issue when we were defining 'dry plant matter' in #18 @kaiiam mentioned the following:

>my issue with this term 'dry matter' is that it's super broad it literally refers to any dry matter, could be anything from dried tomatoes or dry space dust. A more general dry matter class would belong in ENVO or elsewhere then CDNO. I think what you guys mean is dry plant matter. I can understand that you'd prefer to use the label dry matter as that's what your users probably expect to see. I propose if you want to keep the label dry matter but it actually refers to only plant material then maybe keep the def:
Material which remains after the removal of water from plant matter.

@kaiiam Do you think ENVO would like to define this term?

**Comment by maweber-bia** [comment]
Hi, thank you for the answers.

Please do not consider the term "component" in my hierarchy (it is my own top-level). If you consider "nutrient" as a role, then my question is now a little bit misleading. I will wait and see the decision about vitamins (#57 ) 

I just want to add that to me "plant dry matter" and "dry plant matter" are not the same things

If "dry matter" is too generic, you can use "food dry matter" instead and then, plant dry matter is the dry matter content in plants (you can also have dry matter in other foods, this is a measurement of the dried sample where the moisture is not taken into account)

ash + organic matter = 100% of dried sample.
[](url)
see here https://lvscaribouresearch.wordpress.com/about/ashdry-matter/ for illustration

**Comment by kaiiam** [comment]
> If we change it, we might need to contact Uberon as they already have this term 

That's an annotation property from UBERON or PROV which is the multi-species anatomy ontology, I don't see how this is relevant to us or something we should care/worry about. Component isn't defined in that annotation property, so we don't need to use the term. I assumed it was terminology used from the original framework if it's recognizable to people then it's probably worth keeping. 

> Do you think ENVO would like to define this term?

I don't think that's helpful to our purpose here. I'd vote for @maweber-bia's suggestion of using "food dry matter" instead.
